torch/_dynamo/variables/builder.py

```diff
@@ -1516,6 +1516,27 @@ def mark_static_input(self, value: torch.Tensor, guard: bool):
     def wrap_module(self, value: torch.nn.Module):
         from ..eval_frame import OptimizedModule
 
+        # if config.profile_params:
+        metrics_context = get_metrics_context()
+        if metrics_context.in_progress():
+            try:
+                params = (
+                    p
+                    for _, p in value.named_parameters(
+                        recurse=False, remove_duplicate=True
+                    )
+                    if not torch.nn.parameter.is_lazy(p)
+                )
+                metrics_context = get_metrics_context()
+                for p in params:
+                    metrics_context.set_key_value("param_numel_addr", id(p), p.numel())
```

Comment:
I suspect adding a new member to `class MetricsContext`.

```python
class MetricsContext:
    def __init__(self, on_exit: OnExitType):
        self._on_exit = on_exit
        self._metrics: dict[str, Any] = {}
        self._traced_parameters: WeakSet[nn.Parameter] = WeakSet()  # <==
        self._start_time_ns: int = 0
        self._level: int = 0

    def track(self, param: nn.Parameter) -> bool:
        tracked = param in self._traced_parameters
        self._traced_parameters.add(param)
```

Reply:
Switched.

Reply:
FYI - appears paramaters don't work with WeakSets, still had to use id(param).
```
025-03-28T02:01:09.9298631Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/metrics_context.py", line 147, in track
2025-03-28T02:01:09.9299254Z     tracked = param in self._traced_parameters
2025-03-28T02:01:09.9299775Z   File "/opt/conda/envs/py_3.10/lib/python3.10/_weakrefset.py", line 80, in __contains__
2025-03-28T02:01:09.9300268Z     return wr in self.data
2025-03-28T02:01:09.9300893Z torch._dynamo.exc.InternalTorchDynamoError: RuntimeError: Boolean value of Tensor with more than one value is ambiguous
2025-03-28T02:01:09.9301438Z 
```